hazelcast/proxy/reliable_topic.py

```diff
@@ -582,7 +582,7 @@ def publish(self, message: MessageType) -> Future[None]:
             return self._add_or_fail(topic_message)
         elif overload_policy == TopicOverloadPolicy.DISCARD_OLDEST:
             return self._add_or_overwrite(topic_message)
-        elif overload_policy == TopicOverloadPolicy.DISCARD_NEWEST:
+        else:
```

Comment:
Would it make sense to restore `elif overload_policy == TopicOverloadPolicy.DISCARD_NEWEST` here and add an `else` which raises an exception in case an unexpected policy is set?